iceoryx_utils/platform/mac/source/semaphore.cpp

```diff
@@ -0,0 +1,248 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_utils/platform/semaphore.hpp"
+#include "iceoryx_utils/platform/time.hpp"
+
+#include <chrono>
+#include <cstdarg>
+#include <cstdlib>
+#include <iostream>
+#include <thread>
+
+static void deleteValue(std::atomic<int>*& value) noexcept
+{
+    if (value != nullptr)
+    {
+        delete value;
+        value = nullptr;
+    }
+}
+
+iox_sem_t::iox_sem_t() noexcept
+    : m_value{new std::atomic<int>}
```

Comment:
no better solution than creating this on heap?

Reply:
Actually no. iox_sem_t has to be at least movable but std::atomic's are neither copy nor movable therefore we have to put them on the heap.

Reply:
could this be solved with the semaphore pool?
... but I guess it's on the wrong layer